src/app/components/calendar/calendarForm.tsx

```diff
@@ -48,10 +39,17 @@ const CalendarForm = () => {
     const fieldName = e.target.name;
     const fieldValue = e.target.value;
 
-    setCalendar((prevState) => ({
+    setCalendar((prevState: any) => ({
       ...prevState,
       [fieldName]: fieldValue,
     }));
+
+    let newVals = (prevState: any) => {
+      return { ...prevState, [fieldName]: fieldValue };
+    };
+    let values = newVals(calendarSetup);
+
+    localStorage.setItem("calendarData", JSON.stringify(values));
```

Comment:
This place shows one of the reasons having both localStorage and useState is not the best idea. You are setting the localStorage immediately, but setCalendar will modify the state only after some time. So for a small moment, localStorage and local state will contain two different versions of the calendar's state.

I'm not sure if this can lead to bugs (in this case, realistically not because humans do not click that fast, but it shows there is an issue in the memory model 🙂

Reply:
Yeah, resolved in #6  .